docs/updating-components.md

```diff
@@ -0,0 +1,123 @@
+# How to update components?
+
+## Introduction
+
+This is a developer document (not for end users) which explains how to upgrade a particular component in lokoctl. This includes steps to pull config from upstream.
+
+## Find updates
+
+To figure out which component is out dated run the following command in the root of this repository:
+
+```bash
+./scripts/find-updates.sh
+```
+
+## Etcd
+
+Find the old version and newer version from the `./scripts/find-updates.sh` script and export it accordingly.
+
+```bash
+export OLD_VERSION="<old version>"
+export NEW_VERSION="<new version>"
+```
+
+Now run the following commands in the root of this repository:
+
+```bash
+sed -i "s|$OLD_VERSION|$NEW_VERSION|g" assets/lokomotive-kubernetes/aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl
+sed -i "s|$OLD_VERSION|$NEW_VERSION|g" assets/lokomotive-kubernetes/azure/flatcar-linux/kubernetes/cl/controller.yaml.tmpl
+sed -i "s|$OLD_VERSION|$NEW_VERSION|g" assets/lokomotive-kubernetes/bare-metal/flatcar-linux/kubernetes/cl/controller.yaml.tmpl
+sed -i "s|$OLD_VERSION|$NEW_VERSION|g" assets/lokomotive-kubernetes/google-cloud/flatcar-linux/kubernetes/cl/controller.yaml.tmpl
+sed -i "s|$OLD_VERSION|$NEW_VERSION|g" assets/lokomotive-kubernetes/kvm-libvirt/flatcar-linux/kubernetes/cl/controller.yaml.tmpl
+sed -i "s|$OLD_VERSION|$NEW_VERSION|g" assets/lokomotive-kubernetes/packet/flatcar-linux/kubernetes/cl/controller.yaml.tmpl
+make update-assets
+```
+
+- Releases: https://github.com/etcd-io/etcd/releases.
+
+## Calico
+
+To update Calico update the image tags in following files:
+
+- `assets/lokomotive-kubernetes/bootkube/resources/charts/calico/values.yaml`
```

Comment:
I don't like hardcoding all the paths in this document, but I guess it just describes the reality. Can we add some kind of "at time of writing" note on the top, that those path may change in the future?